### PR TITLE
[MM-52119]: Add condition for bot tag for webhook post if a bot account is used for webhook

### PR DESCRIPTION
**Pull request by AshishDhama**
#### Summary
Add condition for bot tag for webhook post if a bot account is used for webhook.

#### Ticket Link
  Fixes https://mattermost.atlassian.net/browse/MM-52119

#### Screenshots
|  before  |  after  |
|----|----|
image | Screenshot 2023-04-12 at 3 08 11 PM |

#### Release Note
```release-note
- Add condition for bot tag for webhook post if a bot account is used for webhook.
```

### CODE REVIEW

File: webapp/channels/src/components/post/user_profile.tsx [hunk 1]
```diff
@@ -95,7 +95,11 @@ const PostUserProfile = (props: Props): JSX.Element | null => {
                 />
             );
 
-            botIndicator = (<BotTag/>);
+            // user profile component checks and add bot tag in case webhook is from bot account, but if webhook is from user account we need this.
```
Comment (harshilsharma63) [inline webapp/channels/src/components/post/user_profile.tsx:98]: Can we add more/better description of why we're doing this? Maybe mention the jira ticket in code comment itself? Without the context of the community thread, the code looks weird - we add bot tag if it's not a bot.

**Comment by nevyangelova** [comment]
> Can we add unit test for this?

@harshilsharma63 i will add unit tests in my spare time, was supposed to be added as part of the post component refactoring by web app team, ticket here: https://mattermost.atlassian.net/browse/MM-52133

**Review by harshilsharma63 — APPROVED** [review]
LGTM 👍

**Comment by harshilsharma63** [comment]
> > Can we add unit test for this?
> 
> @harshilsharma63 i will add unit tests in my spare time, was supposed to be added as part of the post component refactoring by web app team, ticket here: https://mattermost.atlassian.net/browse/MM-52133

In that case, LGTM 👍

**Review by furqanmlk — APPROVED** [review]
Thanks, @AshishDhama for QA steps.
Here are the steps I followed,
1. Create a Bot Account `( note Access Token )`
2. Create a channel `(note channel id)` and add the Bot account created in Step 1
3. run the following curl command 
```
curl --location 'https://<server-url>/api/v4/posts' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer <bot-access-token - see step 1>' \
--data '{"channel_id":"<channel-id - see step 2>", "message":"This is a message from a bot", "props":{
    "from_webhook": "true",
    "attachments": [{"pretext": "Look some text","text": "This is text"}]}}'
 ```
    
 5. Post has only one Bot badge
  
image

### Before PR
image

**Comment by AshishDhama** [comment]
@furqanmlk Thanks a lot adding the steps.

**Pull request merged**